--- tob.rc.dpkg-dist	2004-02-26 21:42:28.000000000 -0500
+++ tob.rc.default	2004-03-06 20:01:39.000000000 -0500
@@ -15,21 +15,43 @@
 # Debian stores tob configuration in and below this directory 
 TOBHOME="/etc/tob"			
 
-# Examples for possible backup devices selected via BACKUPDEV
-#
-# /dev/ftape should be a link to your the device corresponding to your tape
+# Examples for possible backup devices selected via BACKUPDEV:
+# 
+# 1) A device, for example
 #
 # BACKUPDEV="/dev/ftape"			
-#
-# or use the tape drive on another networked box 'groucho.my.net'
+# 
+# where /dev/ftape should be a link to your the device corresponding 
+# to your tape.
+# 
+# Or use the tape drive on another networked box 'groucho.my.net':
+# 
 # BACKUPDEV="groucho.my.net:/dev/ftape"	
 #
-# or use a filesystem-based backup:
-BACKUPDEV="/var/backups/"
-
-# If you're using a filesystem backup, tob will clean out old backups
-# for you once they reach a certain age.  Set that age (in days) here:
-MAXBACKUPAGE=5
+# 2) A directory, for example:
+# 
+# BACKUPDEV="/var/lib/tob"
+# 
+# In this case, tob chooses the file name to use. This is the default.
+# 
+# 3) A file, for example:
+# 
+# if [ ! "$TYPE" == "none" ]; then
+#   TODAYIS=`date +%y%m%d`
+#   MYBACKUPDIR="/Backups/${HOSTNAME}/${TODAYIS}-${TYPE}"
+#   BACKUPDEV="${MYBACKUPDIR}/${TODAYIS}-${TYPE}-${VOLUMENAME}.tar.gz"
+#   mkdir -p ${MYBACKUPDIR}
+# fi
+#
+# BACKUPDEV can be overridden by specifying "-f path" on the tob
+# command line.
+
+# If you're using a filesystem backup, tob can delete old backups
+# in the directory containing the new backup.  It will only delete
+# backups that aren't needed for a full restore, and which are
+# at least MAXBACKUPAGE days old.  If MAXBACKUPAGE is negative, 
+# no old backups are deleted.
+MAXBACKUPAGE=-1
 
 # PRECMD and POSTCMD are not needed for loading/unloading of the ftape
 # modules if the following two lines are added to /etc/conf.modules so that 
@@ -47,9 +69,23 @@
 #
 # I use the following command to retension the tape before I backup:
 # PRECMD='ftmt -f $BACKUPDEV retension 2>/dev/null; echo Tape retensioned'
-#DELETIONSCOMMAND='cat >$BACKUPDIR/${VOLUMENAME}_${DATE}_${TYPE}_deletions'
-# Debian bug report #233300 suggests
-DELETIONSCOMMAND='cat >${TOBLISTS}/${VOLUMENAME}_${DATE}_${TYPE}_deletions'
+
+# A list of all files that have been deleted will be piped to
+# the following command.  This information is necessary after
+# a differential or incremental backup if you want to restore
+# the filesystem to the same state it is in.  Thus this data
+# should be saved along with your backups.  If unset, no
+# deletion list is saved.  Examples:
+# 
+# If $BACKUPDEV is a directory:
+# 
+DELETIONSCOMMAND='cat >${BACKUPDEV}/${VOLUMENAME}_${DATE}_${TYPE}_deletions'
+# 
+# If $BACKUPDEV is a device:
+# 
+# DELETIONSCOMMAND='cat >${BACKUPDEV}'
+#
+# In either case, "cat" could be replaced with "gzip".
 
 # let tob be explicit
 VERBOSE='yes'				
@@ -69,8 +105,10 @@
 GZIPFACTOR='-G 9 -Z '
 
 # These commands select afio as the backup engine with the selected 
-# compression and block sizes, more examples for tob and afio use are 
-# provided in /usr/doc/tob/examples
+# compression and block sizes.  More examples for tob and afio use are 
+# provided in /usr/doc/tob/examples.  Note that if BACKUPDEV is set
+# to a directory, tob will reset it to a filename before calling
+# these commands.
 BACKUPCMD='afio $BLOCKSIZE $BUFFERBLK $GZIPFACTOR -o $BACKUPDEV < $FILELIST'
 BACKUPCMDTOSTDOUT='afio $BLOCKSIZE $BUFFERBLK $GZIPFACTOR -o - < $FILELIST'
 LISTCMD='afio $BLOCKSIZE $BUFFERBLK -t $BACKUPDEV'
@@ -81,7 +119,3 @@
 # Note: if you change this, you might have to mop up backups with the old
 # extension yourself.
 EXT=gz
-
-
-
-
