--- tob	2004-03-06 20:21:52.000000000 -0500
+++ tob.iterate	2004-03-07 21:26:17.000000000 -0500
@@ -493,12 +493,15 @@
 {
     VOLUMENAME=$2		     # store name for usage in .rc file
     if [ -d "$BACKUPDEV" ]; then
-	BACKUPDEV=`$LS ${BACKUPDEV}/${VOLUMENAME}_*full.${EXT}`
+      for i in `$LS ${BACKUPDEV}/${VOLUMENAME}_*full.${EXT}`; do
+      	BACKUPDEV=$i
+	message "Now starting verify program for $BACKUPDEV."
+	(cd /; eval "$VERIFYCMD")
+      done
+    else
+	message "Now starting verify program for $BACKUPDEV."
+	(cd /; eval "$VERIFYCMD")
     fi
-    message "Now starting verify program for $BACKUPDEV."
-    ( cd /;
-      eval "$VERIFYCMD" 
-    )
 }
 
 ############################################################################
